arangod/Aql/RestAqlHandler.h

```diff
@@ -136,6 +136,16 @@ class RestAqlHandler : public RestVocbaseBaseHandler {
   // our query registry
   QueryRegistry* _queryRegistry;
 
+  // Guard that keeps the lock on the QueryContext to serialize
+  // concurrent snippet execution
+  struct QueryGuard {
+    explicit QueryGuard(aql::ExecutionEngine* engine);
```

Comment:
I think we should disable copy construction and copy assignment of `QueryGuard` objects explicitly.
Plus we may want to explicitly define the behavior for move construction or move assignment.